datafusion/src/optimizer/filter_push_down.rs

```diff
@@ -379,6 +379,11 @@ fn optimize(plan: &LogicalPlan, mut state: State) -> Result<LogicalPlan> {
                 }
 
                 if add_to_provider {
+                    // Don't add expression again if it's already present in
+                    // pushed down filters.
+                    if new_filters.iter().any(|filter| filter.eq(filter_expr)) {
```

Comment:
```suggestion
                    if new_filters.contains(filter_expr) {
```
something like this should work?

Reply:
Nice

Reply:
Thanks @Dandandan, I completely forgot about `contains` on `Vec`.